docs/src/index.md

```diff
@@ -0,0 +1,144 @@
+# Bijectors.jl
```

Comment:
@devmotion I'm curious what you think of this design:) You can ignore the code for now, the main thing is just whether or not you're happy with the idea of doing 

``` julia
forward(b, x) = forward_single(b, x)

# Non-batched version
forward_single(b, x) = ...

# Batched version
forward_multiple(b, x) = ...
```

We can then introduce a `Batch` type which generalizes `ColVecs` etc., and do

``` julia
forward(b, x::AbstractBatch) = forward_multiple(b, x)
```

My plan is to split this into two PRs:
1. Remove all "official" support for batched computation, thus always assuming that the input given represents a _single_ input (some bijectors might still support it, but there's not going to be a "official" support for it). In this PR there is no `forward_single`, etc., just `forward`.
2. Add support for batching and overload `forward_single` instead of `forward`.

_But_ ideally we'd fully adopt the ChangesOfVariables.jl interface, i.e. replace `forward` with `with_logabsdet_jacobian`. The issue here is that we of course can no longer do

``` julia
with_logabsdet_jacobian(b, x) = with_logabsdet_jacobian_single(b, x)
```

etc. which makes me want to keep `forward` _or_ ChangesOfVariables decides to take on a similar interface and encourage people to instead implement `with_logabsdet_jacobian_single`.

Thoughts?

Reply:
I think it will be a huge improvement if batching is declared more explicitly. Is there a specific reason why one needs `forward_single` and `forward_multiple` instead of just `forward(::MyBijector, x)` and `forward(::MyBijector, x::AbstractBatch)`? More functions means more entry points and hence more possible confusion for developers. If there's no dedicated `_single` function anymore, it would also work better with the CoV API, I assume?

Reply:
The issue is method ambiguity :confused: If only `forward` or `with_logabsdet_jacobian` is the entry-point, then we cannot provide sane defaults, e.g. `forward(b, x::AbstractBatch)` without every other implementation of `forward` being _very_ explicit.

Reply:
Yeah this is annoying - but there are multiple different options I think. One would be e.g. to implement (possibly different) default implementations but not as `forward(...)` but some `forward_batch_style1(..)` (whatever) and then to "activate" it by defining `forward(b::MyBijector, x::AbstractBatch) = forward_batch_style1(b, x)` etc. I.e., batch support would be enabled manually but without much effort. And one could reduce the amount of code even more with some helper macro.

Reply:
Yeah that works of course, but I think this will be easy to forget vs. always _just_ implemented `*_single` and batching is guaranteed to at least work.

Both approaches feel sub-optimal :confused: